### ISSUE TITLE
About the identification of class F

### ISSUE THREAD

**Issue by PGCJ** [SajadMo/ECG-Heartbeat-Classification-seq2seq-model#14]
Hello author, may I ask why class F is so difficult to identify? I have tried a lot of methods, as well as the code on github, basically have difficulty recognizing F, but which do not open source code can be recognized and recall is also high.

**Comment by andrelopes2001** [comment]
Look into the theory behind what is supposed to be labeled as F, S and V. F is the fusion of ventricular and normal beats.

**Comment by FooliDog** [comment]
> Hello author, may I ask why class F is so difficult to identify? I have tried a lot of methods, as well as the code on github, basically have difficulty recognizing F, but which do not open source code can be recognized and recall is also high.

Hello, the classification experiment I am currently conducting for inter patient is not very effective for S-class classification. Can I refer to your code?My email is diaolisong@outlook.com

**Comment by huinee56** [comment]
> > Hello author, may I ask why class F is so difficult to identify? I have tried a lot of methods, as well as the code on github, basically have difficulty recognizing F, but which do not open source code can be recognized and recall is also high.作者你好，请问为什么F类这么难识别？我尝试了很多方法，以及github上的代码，基本上都很难识别F，但其中没有开源代码的可以识别，召回率也很高。
> 
> Hello, the classification experiment I am currently conducting for inter patient is not very effective for S-class classification. Can I refer to your code?My email is [diaolisong@outlook.com](mailto:diaolisong@outlook.com)你好，我目前正在做的inter - patient分类实验对s类分类效果不是很好。我可以参考一下你的代码吗？我的邮箱是diaolisong@outlook.com

Could you kindly inform me how you train the classification model in the patient-wise mode? I already know that the training set is DS1 and the test set is DS2. Do you use a validation set? If so, how is the validation set obtained? If not, what is your strategy for saving the model?
请问你方便告知我你在患者间模式下是如何对分类模型进行训练的，我已经知晓训练集是DS1,测试集是DS2，请问您有用验证集吗？请问如果用到了，验证集是怎么来的？如果没有用到，请问你保存模型的策略是什么？

**Comment by FooliDog** [comment]
When I was doing experiments and re-enactments, I adopted 90% of DS1 for training, 10% for verification, DS2 for testing. At present, the strategy for saving the model is to completely run all the epochs, and I did not adopt the early-stop strategy.获取Outlook for Android

**Comment by huinee56** [comment]
> When I was doing experiments and re-enactments, I adopted 90% of DS1 for training, 10% for verification, DS2 for testing. At present, the strategy for saving the model is to completely run all the epochs, and I did not adopt the early-stop strategy.获取Outlook for Android

If this operation is performed, there will be a problem: the train set and the validation set may come from the same patient, which does not conform to the meaning of inter-patient.Will this have an impact?

**Comment by PGCJ** [comment]
You maybe get wrong with it. This dataset has 2 divided method, liking intra-patient and inter-patient. The inter-patient is a best way to do research. You can look through my paper, which was discussed it.
---- Replied Message ----
***@***.***>Date6/11/2025 ***@***.***>***@***.***>,
***@***.***>SubjectRe: [MousaviSajad/ECG-Heartbeat-Classification-seq2seq-model] About the identification of class F (Issue #14)
huinee56 left a comment (MousaviSajad/ECG-Heartbeat-Classification-seq2seq-model#14)
When I was doing experiments and re-enactments, I adopted 90% of DS1 for training, 10% for verification, DS2 for testing. At present, the strategy for saving the model is to completely run all the epochs, and I did not adopt the early-stop strategy.获取Outlook for Android
If this operation is performed, there will be a problem: the train set and the validation set may come from the same patient, which does not conform to the meaning of inter-patient.Will this have an impact?

**Comment by huinee56** [comment]
I couldn't find your paper.Could you please provide the link to your paper？Additionally, may I ask if you used a validation set during your experiments? If so, how was it created, and what criteria did you use to select the best model?

**Comment by PGCJ** [comment]
"Enhancing ECG Heartbeat classification with feature fusion neural networks and dynamic minority-biased batch weighting loss function". If it was useful for you, and you can cite it. (lol)
---- Replied Message ----
***@***.***>Date6/12/2025 ***@***.***>***@***.***>,
***@***.***>SubjectRe: [MousaviSajad/ECG-Heartbeat-Classification-seq2seq-model] About the identification of class F (Issue #14)
huinee56 left a comment (MousaviSajad/ECG-Heartbeat-Classification-seq2seq-model#14)
You maybe get wrong with it. This dataset has 2 divided method, liking intra-patient and inter-patient. The inter-patient is a best way to do research. You can look through my paper, which was discussed it.
I couldn't find your paper.Could you please provide the link to your paper？

**Comment by huinee56** [comment]
I read the paper you sent, and in your paper, it only mentioned 'divided the datasets to separate patients into training and test sets,' but I still have some specific questions regarding the experimental details: how was the training performed in the inter-patient mode? Was a validation set used, and if so, where did it come from? What was the best metric used for saving the model? If no validation set was used, were you directly using the test set for validation? In that case, what was the best metric used for saving the model.我阅读了关于您发的论文，在您的论文里只提及了“divided the datasets to separate patients into training and test sets”但是我还是对具体的实验细节有疑问：您在患者间模式下，是如何进行训练的？有用到验证集吗？如果有来自哪里？然后选择保存模型的最佳指标是什么？如果没有用到验证集？你是直接利用测试集即DS2进行验证吗？那么选择保存模型的最佳指标又是什么？

**Comment by PGCJ** [comment]
DS1和DS2是AAMI+luz等人已经设置好了的，绝大部分都是按照这个来的，建议你看看他们的设计详情。在患者间范式（inter-patient）下，DS1为训练集，DS2为测试集，没有验证集。患者内范式（intra-patient）下，我是8:2随机的，有一些工作是7:3（但是患者内范式没有任何意义）。最佳指标看你自己需要什么，比如AUC?sensitivity？都可以，取决你论文想往什么靠，我主要是看sensitivity+accuracy+specificity；
---- Replied Message ----
***@***.***>Date6/12/2025 ***@***.***>***@***.***>,
***@***.***>SubjectRe: [MousaviSajad/ECG-Heartbeat-Classification-seq2seq-model] About the identification of class F (Issue #14)
huinee56 left a comment (MousaviSajad/ECG-Heartbeat-Classification-seq2seq-model#14)
I read the paper you sent, and in your paper, it only mentioned 'divided the datasets to separate patients into training and test sets,' but I still have some specific questions regarding the experimental details: how was the training performed in the inter-patient mode? Was a validation set used, and if so, where did it come from? What was the best metric used for saving the model? If no validation set was used, were you directly using the test set for validation? In that case, what was the best metric used for saving the model.我阅读了关于您发的论文，在您的论文里只提及了“divided the datasets to separate patients into training and test sets”但是我还是对具体的实验细节有疑问：您在患者间模式下，是如何进行训练的？有用到验证集吗？如果有来自哪里？然后选择保存模型的最佳指标是什么？如果没有用到验证集？你是直接利用测试集即DS2进行验证吗？那么选择保存模型的最佳指标又是什么？

**Comment by FooliDog** [comment]
> 你好，作者，请问为什么F类如此难以识别？我尝试过很多方法，还有GitHub上的代码，基本上都很难识别F，但那些不是开源代码的，可以识别，记忆力也很高。

Hello, regarding the research on type F, was your experiment successful?
